platform/android/java/lib/build.gradle

```diff
@@ -56,27 +54,20 @@ android {
         // files is only setup for editing support.
         gradle.startParameter.excludedTaskNames += taskPrefix + "externalNativeBuild" + buildType
 
-        // Create tasks to generate the Godot native libraries.
-        def taskName = "compileGodotNativeLibs" + buildType
-        def releaseTarget = "release"
-        if (buildType == "Debug") {
-            releaseTarget += "_debug"
+        def releaseTarget = supportedTargets[buildType.toLowerCase()]
+        if (releaseTarget == null || releaseTarget == "") {
+            throw new GradleException("Invalid build type: " + buildType)
         }
 
-        def abiTaskNames = []
-        // Creating gradle tasks to generate the native libraries for the supported abis.
-        supportedAbis.each { abi ->
-            def abiTaskName = taskName + abi.capitalize()
-            abiTaskNames += abiTaskName
-            tasks.create(name: abiTaskName, type: Exec) {
-                executable "scons"
-                args "--directory=${pathToRootDir}", "platform=android", "target=${releaseTarget}", "android_arch=${abi}"
-            }
+        if (!supportedAbis.contains(defaultAbi)) {
+            throw new GradleException("Invalid default abi: " + defaultAbi)
         }
 
-        // Creating gradle task to run all of the previously generated tasks.
-        tasks.create(name: taskName, type: GradleBuild) {
-            tasks = abiTaskNames
+        // Creating gradle task to generate the native libraries for the default abi.
+        def taskName = getSconsTaskName(buildType)
+        tasks.create(name: taskName, type: Exec) {
```

Comment:
You still run `scons` here, is it expected?

Reply:
This is a separate workflow from the command line and the two do not collide (I.e: values set here do not affect the command line and vice versa).  
It's set to the `defaultAbi` value and the expectation is that when using and building only from Android studio you should be able to change that value to the desired abi in the root `build.gradle` file where it's defined.
There are some comments where that value is defined that explain its purpose.